hermes/bin/src/event/queue.rs

```diff
@@ -0,0 +1,194 @@
+//! Hermes event queue implementation.
+
+use std::sync::{
+    mpsc::{Receiver, Sender},
+    Arc, Mutex,
+};
+
+use super::{HermesEvent, HermesEventPayload, TargetApp, TargetModule};
+use crate::{
+    app::{HermesAppName, IndexedApps},
+    runtime_extensions::state::State,
+    runtime_state::{HermesRuntimeContext, HermesRuntimeState},
+    wasm::module::{Module, ModuleId},
+};
+
+/// Hermes event queue error
+#[derive(thiserror::Error, Debug, Clone)]
+pub(crate) enum Error {
+    /// Target app not found.
+    #[error("Target app not found, app name: {0:?}.")]
+    AppNotFound(HermesAppName),
+
+    /// Target module not found.
+    #[error("Target module not found, module id: {0:?}.")]
+    ModuleNotFound(ModuleId),
+
+    /// Failed to add event into the event queue. Event queue is closed.
+    #[error("Failed to add event into the event queue. Event queue is closed.")]
+    CannotAddEvent,
+
+    /// Trying to execute one more event execution loop. It is allowed to run only one
+    /// execution loop in a time.
+    #[error("Trying to execute one more event execution loop. It is allowed to run only one execution loop in a time.")]
+    AnotherEventExecutionLoop,
+}
+
+/// Hermes event queue.
+pub(crate) struct HermesEventQueue {
+    /// Hermes event queue sender
+    sender: Sender<HermesEvent>,
+    /// Hermes event queue receiver
+    receiver: Mutex<Receiver<HermesEvent>>,
```

Comment:
Technically yes, but `Receiver` is a part of the HermesQueue itself which could be passed through different threads, because we need to pass `Sender` to another runtime extensions.